alchemy-web/src/routeData.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  defineRouteMiddleware,
+  type StarlightRouteData,
+} from "@astrojs/starlight/route-data";
+
+export const onRequest = defineRouteMiddleware((context) => {
+  let route: StarlightRouteData;
+  // Get the content collection entry for this page.
+  try {
+    route = context.locals.starlightRoute;
+  } catch (_) {
+    // This is a non-starlight route, so we want to skip the og generation
+    return;
+  }
+
+  // Base OG image URL
+  const baseImageUrl = `/og/${route.id ?? "index"}`;
```

Comment:
The logic here needs to be updated. This shouldn't be `/og/` it should be something more like `og.alchemy.run/...`. I don't exactly know how best to get the domain though. @sam-goodwin this is what I was talking about when mentioning it'd be nice to have a config output that didn't require deployment. It's a bit of a chicken and egg situation because you need the config to be able to deploy but you can only get it by doing the deploy?

Reply:
Why do we need a subdomain for og images instead of generating at build time?